### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2021-04-01

### PR DESCRIPTION
**Pull request by TomMilos**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/TomMilos/azure-rest-api-specs/monitor%2FdataCollectionRules%2F2021-04-01%2Fstable...master/).
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionRules_API.json [hunk 1]
```diff
@@ -0,0 +1,854 @@
...
+        },
+        "etag": {
+          "description": "Resource entity tag (ETag).",
+          "type": "string",
+          "readOnly": true,
+          "x-ms-mutability": [
+            "read"
+          ]
+        },
+        "systemData": {
```
Comment (majastrz) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionRules_API.json:777]: >systemData [](start = 9, length = 10)

Do you need to update the examples to reflect the systemData changes?

**Review by majastrz — APPROVED** [review]
Signed off from ARM side with a minor comment about examples.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionRules_API.json (hunk 1 above)
Comment (TomMilos) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionRules_API.json:777]: Thanks, just did that

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionEndpoints_API.json [hunk 2]
```diff
@@ -0,0 +1,576 @@
...
+        },
+        "type": {
+          "description": "The type of the resource.",
+          "type": "string",
+          "readOnly": true,
+          "example": "Microsoft.Insights/dataCollectionEndpoints",
+          "x-ms-mutability": [
+            "read"
+          ]
+        },
```
Comment (weidongxu-microsoft) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionEndpoints_API.json:490]: Usually these will be done by an "allOf"?
https://github.com/Azure/azure-rest-api-specs/blob/e9f792687609adcd8e5d63243349e6b6dbc710a4/specification/monitor/resource-manager/Microsoft.Insights/stable/2020-10-01/activityLogAlerts_API.json#L337-L341

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
@ArcturusZhang Go fails which seems not swagger problem.
@msyyc Python T2 fails which seems not swagger problem.

**Comment by ArcturusZhang** [comment]
The Go breaking changes are accumulated from previous PRs for old api-versions. This PR does not introduce any breaking changes.

**Comment by weidongxu-microsoft** [comment]
still fail on python

**Comment by herreraj-ms** [comment]
Hi, what's missing for approval?
SDK generations (.Net & Phyton) have a dependency on this one.
On the .NET (https://github.com/Azure/azure-sdk-for-net/pull/21128) side a manual fix is needed because specs within Azure Monitor uses ErrorResponse version 1 or 2.

FYI: @TomMilos @Aniruddha1248

**Comment by msyyc** [comment]
For python, the breaking change is expected.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionEndpoints_API.json [hunk 3]
```diff
@@ -0,0 +1,576 @@
...
+      "properties": {
+        "description": {
+          "description": "Description of the data collection endpoint.",
+          "type": "string"
+        },
+        "immutableId": {
+          "description": "The immutable ID of this data collection endpoint resource. This property is READ-ONLY.",
+          "type": "string",
+          "example": "dce-b74e0d383fc9415abaa584ec41adece3"
+        },
```
Comment (kairu-ms) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionEndpoints_API.json:373]: Why this property is not marked as read-only?

Comment (kairu-ms) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2021-04-01/dataCollectionEndpoints_API.json:373]: @TomMilos is it mutable?

**Pull request merged**